docs/source/quick_start.rst

```diff
@@ -0,0 +1,95 @@
+.. _quick-start-section:
+
+Quick Start
+===========
+
+As previously mentioned, ``HookMan`` is a python application that uses plugins written in C/C++.
+
+In order to integrate this project in your application, it's necessary to create a configuration object named :ref:`hook-specs-api-section`. 
+This object provides pieces of information related to which hooks are available and which arguments are expected to be sent or received.
+
+The block code below exemplifies a valid :ref:`hook-specs-api-section` configuration:
+
+
+.. code-block:: python
+
+    from hookman.hooks import HooksSpecs
+
+    def env_temperature(arg1: 'float', arg2: 'float') -> 'float':
+        """
+        Docs for Environment Temperature
+        """
+
+    specs = HooksSpecs(
+        project_name='Acme',
+        version='1',
+        pyd_name='_alfasim_hooks',
+        hooks=[
+            env_temperature,
+        ]
+    )
+
+With the :ref:`hook-specs-api-section` defined, it's possible to generate the necessary files to interact between the application and the plugins implementation.
+
+.. code-block:: python
+
+    # Initializing a class 
+    hook_gen = HookManGenerator(hook_spec_file_path=Path('hook_specs.py'))
+    hook_gen.generate_project_files(dst_path=<dst_dir>)
+
+
+The output from the command above will be the following files:
+
+- HookCaller.hpp
+- HookCallerPython.cpp
+- CMakeLists.txt
+
+These files contain all the code necessary to make the project ``PyBind11_`` integrates with your application, and the CMakeLists file contains a boilerplate
+to compile and generate an ``pyd`` file.
```

Comment:
to compile and generate a binary extension (``.pyd`` or ``.so`` file).

Reply:
I do agree with the sentence to compile and generate a binary extension (.pyd file)
But I didn't understand the mention to the .so since this is not a shared library.